src/utils/CachingUtils.ts

```diff
@@ -1,7 +1,42 @@
+import { DEFAULT_CACHING_SAFE_LAG, DEFAULT_CACHING_TTL } from "../constants";
+import { CachingMechanismInterface, Deposit, Fill } from "../interfaces";
 import { assert } from "./LogUtils";
+import { composeRevivers, objectWithBigNumberReviver } from "./ReviverUtils";
+import { isDefined } from "./TypeGuards";
 
 export function shouldCache(eventTimestamp: number, latestTime: number, cachingMaxAge: number): boolean {
   assert(eventTimestamp.toString().length === 10, "eventTimestamp must be in seconds");
   assert(latestTime.toString().length === 10, "eventTimestamp must be in seconds");
   return latestTime - eventTimestamp >= cachingMaxAge;
 }
+
+/**
+ * Calls the cache's get method and returns the result if it is defined, otherwise returns undefined.
+ * @param key The key to get from the cache.
+ * @param cache The cache to get the key from.
+ * @returns The value associated with the key in the cache, or undefined if the key is not in the cache.
+ */
+export async function getDepositInCache(key: string, cache: CachingMechanismInterface): Promise<Deposit | undefined> {
```

Comment:
These are both copied from relayer-v2?

Reply:
These are copied from `relayer-v2` with the exception that it was originally `redisClient: RedisClient` and now it's `cache: CachingMechanismInterface`